store/mongo/migration_1_2_15.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package mongo
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/mendersoftware/go-lib-micro/mongo/migrate"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	mopts "go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/mendersoftware/deployments/model"
+)
+
+type migration_1_2_15 struct {
+	client *mongo.Client
+	db     string
+}
+
+// Up intrduces special representation of artifact provides and index them.
+func (m *migration_1_2_15) Up(from migrate.Version) error {
+	ctx := context.Background()
+	c := m.client.Database(m.db).Collection(CollectionImages)
+	cr := m.client.Database(m.db).Collection(CollectionReleases)
+
+	// create index for release name
+	storage := NewDataStoreMongoWithClient(m.client)
+	if err := storage.EnsureIndexes(
+		m.db,
+		CollectionReleases,
+		IndexReleaseName,
+	); err != nil {
+		return err
+	}
+
+	cursor, err := c.Find(ctx, bson.M{})
+	if err != nil {
+		return err
+	}
+	defer cursor.Close(ctx)
+	var a model.Image
+	for cursor.Next(ctx) {
+		err = cursor.Decode(&a)
+		if err != nil {
+			break
+		}
+
+		opt := &mopts.UpdateOptions{}
+		upsert := true
+		opt.Upsert = &upsert
+		update := bson.M{
+			"$set": bson.M{
+				StorageKeyReleaseName:     a.ArtifactMeta.Name,
+				StorageKeyReleaseModified: time.Now(),
+			},
+			"$push": bson.M{StorageKeyReleaseArtifacts: a},
+		}
```

Comment:
I was focusing on simplicity here and since the migration can be done in the background, what if someone will upload new artifact while the migration is running?

Reply:
I'm not sure I understand your last point. Migrations are not re-entrant so if the service cannot handle artifact uploads happening in parallel then there will be data inconsistencies.

Reply:
sorry, you're right; new service will update releases during artifacts uploads, so we don't need to look at them when running the migration;
I'll try to optimize the update query

Reply:
ehh, creating release with many artifacts will not work for other reason, the one we're discussing [here](https://github.com/mendersoftware/deployments/pull/867/files#r1229341321)